insecure/integration/functional/test/gossipsub/rpc_inspector/validation_inspector_test.go

```diff
@@ -1079,36 +1113,243 @@ func TestValidationInspector_UnstakedNode_Detection(t *testing.T) {
 	require.Equal(t, uint64(1), invPruneNotifCount.Load())
 }
 
-func randomClusterPrefixedTopic() channels.Topic {
-	return channels.Topic(channels.SyncCluster(flow.ChainID(fmt.Sprintf("%d", rand.Uint64()))))
-}
+// TestValidationInspector_InspectIWants_DuplicateMessageIDs ensures that expected invalid control message notification is disseminated when iWant control messages containing
+// duplicate messages Ids are encountered during validation.
+func TestValidationInspector_InspectIWants_DuplicateMessageIDs(t *testing.T) {
```

Comment:
I'm seeing flaky behavior for this test.
⚠️ Please fix this before merging.

```
=== RUN   TestValidationInspector_InspectIWants_DuplicateMessageIDs
=== PAUSE TestValidationInspector_InspectIWants_DuplicateMessageIDs
=== CONT  TestValidationInspector_InspectIWants_DuplicateMessageIDs
    validation_inspector_test.go:1139: 
        	Error Trace:	/Users/misha/dev/flow-go-flaky-tests/insecure/integration/functional/test/gossipsub/rpc_inspector/validation_inspector_test.go:1139
        	            				/Users/misha/dev/flow-go-flaky-tests/network/p2p/inspector/validation/control_message_validation_inspector.go:712
        	            				/Users/misha/dev/flow-go-flaky-tests/network/p2p/inspector/validation/control_message_validation_inspector.go:425
        	            				/Users/misha/dev/flow-go-flaky-tests/engine/common/worker/worker_builder.go:130
        	            				/Users/misha/dev/flow-go-flaky-tests/module/component/component.go:255
        	            				/usr/local/go/src/runtime/asm_arm64.s:1172
        	Error:      	Should be true
        	Test:       	TestValidationInspector_InspectIWants_DuplicateMessageIDs
    unittest.go:213: 
        	Error Trace:	/Users/misha/dev/flow-go-flaky-tests/utils/unittest/unittest.go:213
        	            				/Users/misha/dev/flow-go-flaky-tests/insecure/integration/functional/test/gossipsub/rpc_inspector/validation_inspector_test.go:1206
        	Error:      	could not close done channel on time: failed to inspect RPC messages on time
        	Test:       	TestValidationInspector_InspectIWants_DuplicateMessageIDs
    pub_sub_tracer.go:171: PASS:	WasIHaveRPCSent(mock.AnythingOfTypeArgument)
    pub_sub_tracer.go:171: PASS:	LastHighestIHaveRPCSize()
    gossip_sub_inspector_notification_distributor.go:83: PASS:	Start(string)
    gossip_sub_inspector_notification_distributor.go:83: PASS:	Ready()
    gossip_sub_inspector_notification_distributor.go:83: PASS:	Done()
    gossip_sub_inspector_notification_distributor.go:83: FAIL:	Distribute(string)
        		at: [/Users/misha/dev/flow-go-flaky-tests/insecure/integration/functional/test/gossipsub/rpc_inspector/validation_inspector_test.go:1401 /Users/misha/dev/flow-go-flaky-tests/insecure/integration/functional/test/gossipsub/rpc_inspector/validation_inspector_test.go:1155]
    gossip_sub_inspector_notification_distributor.go:83: FAIL: 3 out of 4 expectation(s) were met.
        	The code you are testing needs to make 1 more call(s).
        	at: [/usr/local/go/src/testing/testing.go:1150 /usr/local/go/src/testing/testing.go:1328 /usr/local/go/src/testing/testing.go:1570 /usr/local/go/src/runtime/panic.go:522 /usr/local/go/src/testing/testing.go:980 /Users/misha/dev/flow-go-flaky-tests/utils/unittest/unittest.go:213 /Users/misha/dev/flow-go-flaky-tests/insecure/integration/functional/test/gossipsub/rpc_inspector/validation_inspector_test.go:1206]
    identity_provider.go:95: PASS:	ByPeerID(peer.ID)
    identity_provider.go:95: PASS:	ByPeerID(peer.ID)
--- FAIL: TestValidationInspector_InspectIWants_DuplicateMessageIDs (3.15s)
```

Reply:
https://github.com/onflow/flow-go/pull/4574/commits/897cb98946d715d25a9048b7cee53cd9770c2498